File: tests/tainting_rules/ruby/switch.rb
def f()
  i = source()
  #ruleid: test-switch
  sink(i)
  case x
  when 0
    i = sanitize(i)
    #OK:
    sink(i)
  when 1
    unrelated_call()
    #ruleid: test-switch
    sink(i)
  else
    unrelated_call()
    #ruleid: test-switch
    sink(i)
  end
end

File: tests/tainting_rules/php/switch.php
<?php

function f($arg) {
    switch ($arg) {
        case 0:
            $a = 0;
            break;
            //OK:
            sink($source);
        case 1:
            $b = 0;
        case 2:
            $c = 0;
            //ruleid: test-switch
            sink($source);
            break;
        default:
            $d = 0;
    }
}
